### PR TITLE
Migrate Localization Strings files to a String Catalogs file

### LINKED ISSUE #67: Migrate localizations to Xcode 15 String Catalogs
Starting with Xcode 15 we have a better way to manage hardcoded strings and localizations.
It's recommended to migrate the project to String Catalogs.

### PR DESCRIPTION
**Pull request by iuliagroza**
### What does it fix?
Closes #67.

### Context
String Catalogs were introduced with XCode 15. They are intended to solve some of the main issues that arise with using Strings, such as the need for collecting the hardcoded Strings files in a centralised single Strings file (often manually), the lack of ways to measure the extent and to vary localization by device type. More at: https://developer.apple.com/documentation/Xcode/localizing-and-varying-text-with-a-string-catalog.

Therefore, I automatically migrated the Localization Strings files of Teacher Workout to a String Catalog.

### CODE REVIEW

**Review by Utwo — APPROVED** [review]
🔥

File: Teacher Workout/Localization/Localizable.xcstrings [hunk 1]
```diff
@@ -0,0 +1,962 @@
...
+        "en" : {
+          "stringUnit" : {
+            "state" : "translated",
+            "value" : "Share"
+          }
+        },
+        "ro" : {
+          "stringUnit" : {
+            "state" : "translated",
+            "value" : "Share"
```
Comment (danielferecatu) [inline Teacher Workout/Localization/Localizable.xcstrings]:
```suggestion
            "value" : "Partajează"
```
Maybe we could update this as well :)

File: Teacher Workout/Localization/Localizable.xcstrings [hunk 2]
```diff
@@ -0,0 +1,962 @@
...
+        "en" : {
+          "stringUnit" : {
+            "state" : "translated",
+            "value" : "Log out"
+          }
+        },
+        "ro" : {
+          "stringUnit" : {
+            "state" : "translated",
+            "value" : "Log out"
```
Comment (danielferecatu) [inline Teacher Workout/Localization/Localizable.xcstrings]:
```suggestion
            "value" : "Delogare"
```
Another translation we could improve.

**Review by danielferecatu — COMMENTED** [review]
The migration looks good to me and I added 2 suggestions for the Romanian translations.

File: Teacher Workout/Localization/Localizable.xcstrings (hunk 1 above)
Comment (iuliagroza) [reply Teacher Workout/Localization/Localizable.xcstrings]: Great point! Thanks for improving the translations!

**Comment by iuliagroza** [comment]
> The migration looks good to me and I added 2 suggestions for the Romanian translations.

Great point! Thank you for improving the translations!

**Review by danielferecatu — APPROVED** [review]

**Review by MihaiPantiru — APPROVED** [review]

**Comment by goiaalexia** [comment]
very needed, thank you @iuliagroza ! 🔥🔥🔥

**Pull request merged**